File: mexicanWave.js
function wave(str){
    let results = [];
    let count = 0;
    let len = str.split('').length;
    while(count <= (len-1)){
    let strRR = str.split('').map((x,index)=>{ 
    return index==count ? x.toUpperCase() : x
    })
    count++
    newWrd = strRR.join('')
      if(/[A-Z]/.test(newWrd)){
      results.push(newWrd)
     }
    }
    return results;
    }

console.log(wave('hello'))
console.log(wave('codewars'))
console.log(wave('good morning'))
console.log(wave('map'))

File: mexican_wave.py
import re
def wave(people):
   wave_num = 0
   waves = list(people)
   results = []
   while wave_num < len(people):
    alt_wave = list(map(lambda x: x.lower(), waves))
    alt_wave[wave_num] = alt_wave[wave_num].upper()
    results.append(''.join(alt_wave))
    wave_num += 1
   edited_list = list(map(lambda x: x if re.search(r"[A-Z]", x) else '', results))
   return [x for x in edited_list if x]
        
      

print(wave('codewars')) # ['Codewars', 'cOdewars', 'coDewars', 'codEwars', 'codeWars', 'codewArs', 'codewaRs', 'codewarS']
print(wave('hello')) # ['Hello', 'hEllo', 'heLlo', 'helLo', 'hellO']
print(wave('bat man')) # ['Bat man', 'bAt man', 'baT man', 'bat man', 'bat Man', 'bat mAn', 'bat maN']
print(wave('bat')) # ['Bat', 'bAt', 'baT']